src/pages/ProfilesExplorerView/components/SceneProfilesExplorer/SceneProfilesExplorer.tsx

```diff
@@ -419,9 +327,12 @@ export class SceneProfilesExplorer extends SceneObjectBase<SceneProfilesExplorer
     const { explorationType, controls, body } = this.useState();
 
     const [timePickerControl, refreshPickerControl] = controls as [SceneObject, SceneObject];
+    const dataSourceVariable = this.state.$variables!.state!.variables[0] as ProfilesDataSourceVariable;
 
-    const { variables, gridControls } = this.getVariablesAndGridControls(explorationType as ExplorationType);
-    const [dataSourceVariable, ...sceneVariables] = variables as SceneVariable[];
+    const { variables: sceneVariables, gridControls } = (body?.state.primary as any).getVariablesAndGridControls() as {
```

Comment:
TODO in a future PR: enforce `getVariablesAndGridControls` to be a method that exists and returns the adequate object.